### PR TITLE
TreeNode check added for heap.py

### PR DESCRIPTION
**Pull request by HarsheetKakar**
#### Brief description of what is fixed or changed
Doc string of heap.py is updated.
relevant test added.

#### Other comments
Previous PR was messed up, this one should work

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -64,3 +64,4 @@ __pycache__/
 .idea/
 build/
 dist/
+venv/
```
Comment (czgdp1807) [inline .gitignore]: Please add a new line at the end of the file.

File: pydatastructs/trees/heaps.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ class DHeap(Heap):
 
     elements : list, tuple
         Optional, by default 'None'.
-        List/tuple of initial elements in Heap.
+        List/tuple of initial TreeNodes in Heap.
```
Comment (czgdp1807) [inline pydatastructs/trees/heaps.py]:
```suggestion
        List/tuple of initial TreeNode in Heap.

```

File: pydatastructs/trees/heaps.py [hunk 3]
```diff
@@ -83,6 +83,12 @@ def __new__(cls, elements=None, heap_property="min", d=4):
             raise ValueError("%s is invalid heap property"%(heap_property))
         if elements is None:
             elements = DynamicOneDimensionalArray(TreeNode, 0)
+        else:
+            #to check if all elements provided are TreeNodes
+            check_node = map(lambda x: type(x)==type(TreeNode(1,1)),elements)
+            is_node = all(check_node)
+            if(not is_node):
+                raise TypeError("Elements have to be list/tuple of TreeNode")
```
Comment (czgdp1807) [inline pydatastructs/trees/heaps.py]:
```suggestion
            if not all(map(lambda x: _check_type(x, TreeNode), elements)):
                raise TypeError("Elements have to be list/tuple of TreeNode")
```

File: pydatastructs/trees/tests/test_heaps.py [hunk 4]
```diff
@@ -29,8 +29,6 @@ def test_BinaryHeap():
     assert max_heap.extract().key == 100
 
     expected_sorted_elements = [36, 25, 19, 17, 7, 3, 2, 1]
-    l = max_heap.heap[0].left
-    l = max_heap.heap[0].right
```
Comment (czgdp1807) [inline pydatastructs/trees/tests/test_heaps.py]: Why?

File: pydatastructs/trees/tests/test_heaps.py [hunk 5]
```diff
@@ -48,6 +46,15 @@ def test_BinaryHeap():
...
 
+    #To test ValueError on non TreeNode list
+    non_TreeNode_elements = [
+                (7, 7), TreeNode(25, 25), TreeNode(100, 100),
+                TreeNode(1, 1), (2, 2), TreeNode(3, 3),
+                TreeNode(17, 17), TreeNode(19, 19), TreeNode(36, 36)
+            ]
+    assert raises(TypeError, lambda:
+                BinaryHeap(elements = non_TreeNode_elements, heap_property='min'))
+
```
Comment (czgdp1807) [inline pydatastructs/trees/tests/test_heaps.py]:
```suggestion
    assert raises(ValueError, lambda:
                BinaryHeap(elements = non_TreeNode_elements, heap_property='min'))
```

File: pydatastructs/trees/tests/test_heaps.py [hunk 6]
```diff
@@ -48,6 +46,15 @@ def test_BinaryHeap():
     sorted_elements = [min_heap.extract().key for _ in range(8)]
     assert expected_sorted_elements == sorted_elements
 
+    #To test ValueError on non TreeNode list
```
Comment (czgdp1807) [inline pydatastructs/trees/tests/test_heaps.py]:
```suggestion
```

File: pydatastructs/trees/heaps.py [hunk 7]
```diff
@@ -83,6 +83,12 @@ def __new__(cls, elements=None, heap_property="min", d=4):
             raise ValueError("%s is invalid heap property"%(heap_property))
         if elements is None:
             elements = DynamicOneDimensionalArray(TreeNode, 0)
+        else:
+            #to check if all elements provided are TreeNodes
```
Comment (czgdp1807) [inline pydatastructs/trees/heaps.py]:
```suggestion
```

File: pydatastructs/trees/tests/test_heaps.py (hunk 4 above)
Comment (HarsheetKakar) [reply pydatastructs/trees/tests/test_heaps.py]: I might have accidentally removed them. It was showing a warning if I remember correctly

Comment (czgdp1807) [reply pydatastructs/trees/tests/test_heaps.py]: I think the first line can be removed safely, though do it in a different PR.

Comment (HarsheetKakar) [reply pydatastructs/trees/tests/test_heaps.py]: The variable is never being used, so idk whats the point of having it.

Comment (czgdp1807) [reply pydatastructs/trees/tests/test_heaps.py]: Sure, feel free to remove it in a separate PR.

**Pull request merged**